rust/connlib/tunnel/src/client.rs

```diff
@@ -621,24 +550,28 @@ impl ClientState {
         self.gateways_site.insert(gateway_id, site_id);
 
         if self.peers.get(&gateway_id).is_some() {
-            self.peers.add_ips_with_resource(
-                &gateway_id,
-                &self.get_resource_ip(desc, &awaiting_connection.domain),
-                &resource_id,
-            );
+            self.peers
+                .add_ips_with_resource(&gateway_id, &ips, &resource_id);
 
             return Ok(Some(Request::ReuseConnection(ReuseConnection {
                 resource_id,
                 gateway_id,
-                payload: awaiting_connection.domain,
+                payload: awaiting_connection_details.domain,
             })));
         };
 
+        self.peers.insert(
+            GatewayOnClient::new(gateway_id, &ips, HashSet::from([resource_id])),
+            &[],
+        );
```

Comment:
@conectado Because we need access to `awaiting_connection_details` (which we remove in this function now) in order to create the `GatewayOnClient` in `peers`, I had to move the creation of the peer into this function.